lib/ecto/schema.ex

```diff
@@ -1771,11 +1771,15 @@ defmodule Ecto.Schema do
 
   @doc false
   def __field__(mod, name, type, opts) do
-    check_field_type!(name, type, type, opts)
+    # type = convert(type)
+    type = check_field_type!(mod, name, type, type, opts)
     Module.put_attribute(mod, :changeset_fields, {name, type})
     define_field(mod, name, type, opts)
   end
 
+  # defp convert(:enum), do: Ecto.Type.Enum
```

Comment:
Btw, in the future we will simply call it `Ecto.Enum` and we won't need to convert this. We already do this for `Ecto.UUID`, so it is fine. :) So you can remove the commented out code!

Reply:
Done. `Ecto.Type.Enum` and `:enum` are both now `Ecto.Enum`